### ISSUE TITLE
Loading UTF8 content is not possible with HTML reader class

### ISSUE THREAD

**Issue by T00by1** [PHPOffice/PhpSpreadsheet#866]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
UTF8 HTML content loaded into the spreadsheet.

### What is the current behavior?
Failure to load because the validation of tags cannot be completed.
startsWithTag needs to check for BOM when looking for first character in document.

### What are the steps to reproduce?
Use Simple example 46. Open template file and add BOM. Retry example.

### Which versions of PhpSpreadsheet and PHP are affected?
"phpoffice/phpspreadsheet": "1.6.0"
 "php": ">=7.1.20"

Suggested simple solution in startsWithTag :
Make function not static and add 
```php
if($this->inputEncoding == 'UTF-8') {
    if(substr($data, 0, 3) === chr(0xEF) . chr(0xBB) . chr(0xBF)) {
      $data = substr($data, 3);
     }
}
```

**Comment by MarkBaker** [comment]
HTML markup shouldn't be identified as UTF-8 by a BOM, but by `<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">` in the `header` block of the markup

**Comment by T00by1** [comment]
I would respectfully like to disagree. 
https://www.w3.org/TR/html5/syntax.html states that HTML documents can have an optional BOM at the start. Also a note further down relates to how this will work when Content-Type is present.
"A leading Byte Order Mark (BOM) causes the character encoding argument to be ignored and will itself be skipped."

**Comment by T00by1** [comment]
> I would respectfully like to disagree.
> https://www.w3.org/TR/html5/syntax.html states that HTML documents can have an optional BOM at the start. So detecting and successfully parse HTML with BOM should be added further down the road as well.

**Comment by oleibman** [comment]
Fixed by PR #4019.
